src/index.js

```diff
@@ -62,25 +65,31 @@ var contourId = 0;
 
 function processPolygon(polygon, isSubject, depth, queue, bbox) {
   var i, len;
-  if (typeof polygon[0][0] === 'number') {
-    for (i = 0, len = polygon.length - 1; i < len; i++) {
-      processSegment(polygon[i], polygon[i + 1], isSubject, depth + 1, queue, bbox);
-    }
-  } else {
-    for (i = 0, len = polygon.length; i < len; i++) {
-      contourId++;
-      processPolygon(polygon[i], isSubject, contourId, queue, bbox);
-    }
+  for (i = 0, len = contourOrHole.length - 1; i < len; i++) {
+    processSegment(contourOrHole[i], contourOrHole[i + 1], isSubject, depth + 1, queue, bbox, isExteriorRing);
   }
 }
 
-
 function fillQueue(subject, clipping, sbbox, cbbox) {
   var eventQueue = new Queue(null, compareEvents);
-  contourId = 0;
-
-  processPolygon(subject,  true,  0, eventQueue, sbbox);
-  processPolygon(clipping, false, 0, eventQueue, cbbox);
+  subject.forEach(function (contour, counterIndex) {
```

Comment:
so here you are absolutely sure that it's a polygon, not a multipolygon? not sure from the code